x/finality/keeper/liveness.go

```diff
@@ -200,3 +182,21 @@ func (k Keeper) updateSigningInfo(
 
 	return modifiedSignInfo, &signInfo, nil
 }
+
+func (k Keeper) jailSluggishFinalityProvider(ctx context.Context, fpBtcPk *types.BIP340PubKey) error {
+	err := k.hooks.AfterSluggishFinalityProviderDetected(ctx, fpBtcPk)
+	if err != nil {
+		return err
+	}
+
+	err = sdk.UnwrapSDKContext(ctx).EventManager().EmitTypedEvent(
+		finalitytypes.NewEventJailedFinalityProvider(fpBtcPk),
+	)
+	if err != nil {
+		return fmt.Errorf("failed to emit sluggish finality provider detected event: %w", err)
+	}
+
+	finalitytypes.IncrementSluggishFinalityProviderCounter()
```

Comment:
Checked and couldn't find the call to the corresponding 'Decrement` call. Perhaps just remove this counter?